docker/Dockerfile.llvm-project

```diff
@@ -22,49 +22,49 @@ WORKDIR ${WORK_DIR}
-              autoconf automake ca-certificates cmake diffutils \
-              file java-11-openjdk-devel java-11-openjdk-headless \
-              gcc gcc-c++ git libtool make ncurses-devel \
-              python39 python39-devel python39-numpy python39-pip \
-              python39-setuptools python39-wheel zlib-devel && \
-          # Use same versions as those in ubuntu:focal
-          pip3 install -q \
-               Cython pytest==4.6.9 pytest-forked==1.1.3 \
-               pytest-xdist==1.31.0 typing-extensions==3.7.4.1 && \
-          rm -rf /var/cache/dnf/* && \
-          # Install ninja
-          git clone -b v1.10.2 https://github.com/ninja-build/ninja.git && \
-          cd ninja && mkdir -p build && cd build && \
-          cmake .. && \
-          make -j${NPROC} install && \
-          cd ${WORK_DIR} && rm -rf ninja; \
-       fi \
+    DEBIAN_FRONTEND=noninteractive && \
+    apt-get update -qq && \
+    apt-get install -qq -y --no-install-recommends tzdata && \
+    ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+    dpkg-reconfigure -f noninteractive tzdata && \
+    apt-get install -qq -y --no-install-recommends \
+    autoconf automake ca-certificates cmake curl \
+    default-jdk-headless gcc g++ git libncurses-dev \
+    libtool make maven ninja-build openjdk-11-jdk-headless \
+    python3 python3-dev python3-distutils python3-numpy \
+    python3-pip python3-pytest-xdist python3-setuptools \
+    python3-typing-extensions zlib1g-dev && \
+    rm -rf /var/lib/apt/lists/* && \
+    ln -sf /usr/bin/pytest-3 /usr/bin/pytest; \
+    elif [ "${distro}" = "rhel" ] || [ "${distro}" = "fedora" ]; then \
+    ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+    ([ -x /usr/bin/microdnf ] && microdnf install -y yum) && \
+    yum install -q -y \
+    https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+    yum update -q -y && \
+    yum install -q -y \
+    autoconf automake ca-certificates cmake diffutils \
+    file java-11-openjdk-devel java-11-openjdk-headless \
+    gcc gcc-c++ git libtool make ncurses-devel \
+    python39 python39-devel python39-numpy python39-pip \
+    python39-setuptools python39-wheel zlib-devel && \
+    # Use same versions as those in ubuntu:focal
+    pip3 install -q \
+    Cython pytest==4.6.9 pytest-forked==1.1.3 \
+    pytest-xdist==1.31.0 typing-extensions==3.7.4.1 && \
+    rm -rf /var/cache/dnf/* && \
+    # Install ninja
+    git clone -b v1.10.2 https://github.com/ninja-build/ninja.git && \
+    cd ninja && mkdir -p build && cd build && \
+    cmake .. && \
+    make -j${NPROC} install && \
+    cd ${WORK_DIR} && rm -rf ninja; \
+    fi \
     && ln -sf /usr/bin/python3 /usr/bin/python
 
 # Install protobuf
-ARG PROTOBUF_VERSION=3.18.3
-RUN git clone -b v${PROTOBUF_VERSION} --recursive https://github.com/google/protobuf.git \
+ARG PROTOBUF_VERSION=3.20.3
+RUN git clone -b v${PROTOBUF_VERSION} --recursive https://github.com/protocolbuffers/protobuf.git \
```

Comment:
This should be the only change to Dockerfile.llvm-project. The other messed up indentation changes should be reverted.

Reply:
I meant the version and url changes should be the only changes needed.

Reply:
> I meant the version and url changes should be the only changes needed.

Sure!! Let me change the format

Reply:
@gongsu832 How does this look now? Ready to merge this in :)